folding-schemes/src/arith/r1cs.rs

```diff
@@ -138,9 +221,24 @@ pub fn extract_w_x<F: PrimeField>(cs: &ConstraintSystem<F>) -> (Vec<F>, Vec<F>)
 #[cfg(test)]
 pub mod tests {
     use super::*;
-    use crate::utils::vec::tests::{to_F_matrix, to_F_vec};
+    use crate::{
+        commitment::pedersen::Pedersen,
+        utils::vec::tests::{to_F_matrix, to_F_vec},
+    };
 
-    use ark_pallas::Fr;
+    use ark_pallas::{Fr, Projective};
+
+    #[test]
+    pub fn sample_relaxed_r1cs() {
+        let rng = rand::rngs::OsRng;
+        let r1cs = get_test_r1cs::<Fr>();
+        let (prover_params, _) = Pedersen::<Projective>::setup(rng, r1cs.A.n_rows).unwrap();
+
+        let relaxed_r1cs = r1cs.relax();
+        let sampled =
+            relaxed_r1cs.sample::<Projective, Pedersen<Projective, true>>(&prover_params, rng);
+        assert!(sampled.is_ok());
```

Comment:
could it be that this is only checking that the `.sample()` method does not return an error, but it is not testing if the generated rel-r1cs satisfies the relation?

Reply:
There is a `debug_assert` in the `sample()` method. I'm assuming that the only thing that could fail when sampling is at computing `E` (which does return an error). However, passed that step I don't think there should be any particular reason for the sampled instance to not be satisfiable?  Hence a `debug_assert` only running for non optmized builds. Or do you see it some other way?

Reply:
Oh right! I missed the debugassert in the sample circuit! Then it makes sense as it is! Thanks!